### ISSUE TITLE
Username or Email is only checked for duplication, not validity, at Request New Password

### ISSUE THREAD

**Issue by alexfdietz** [HTBox/crisischeckin#359]
Navigate to the "Forgot Your password" screen.  
Typed in a ridiculous username: 11111001110 that doesn't exist.
Clicked on Request new password.  
System responded with "password reset request sent" even though the user doesn't exist. 

If I type in an existing user (alexfdietz) the system responds with the exception noted in an earlier issue (so the system is clearly checking the usernameentry against the database).  If I leave the field blank the system responds with "the username or email fieldis required".  

this bug was found during the StarWest virtual conference.

**Comment by BillWagner** [comment]
We should fix this.
However, I want to make sure that we don't provide clues to hackers that emails do exist.

The behavior of saying "we sent the request" is correct, even if the email does not exist in the DB. That way, hackers don't learn what accounts exist. The exception is clearly a bug.

**Comment by tonysurma** [comment]
agree.  The common practice we should follow is basically "If there is an account associated with that email address/user id a password reset request has been sent... blah blah blah"
